profiling/src/profile/mod.rs

```diff
@@ -1395,51 +1612,28 @@ mod api_test {
 
         // The trace endpoint label should be added to the first sample
         assert_eq!(s1.labels.len(), 3);
+        let mut labels: Vec<_> = s1
+            .labels
+            .iter()
+            .map(|l| {
+                let k = serialized_profile.string_table.get(l.key as usize).unwrap();
+                let s = serialized_profile.string_table.get(l.str as usize).unwrap();
+                (k, s, l.num)
+            })
+            .collect();
+        labels.sort();
```

Comment:
I realize the existing test already did this but... relying on `sort` order here seems kinda brittle :sweat_smile:.

Reply:
Yeah.  We should fix the broader set of these tests, which are indeed kinda brittle